main/logkeeper.go

```diff
@@ -69,34 +71,47 @@ func main() {
 	n.Use(gzip.Gzip(gzip.DefaultCompression))
 	n.UseHandler(gorillaCtx.ClearHandler(router))
 
-	serviceWait := make(chan struct{})
-	service := getService(fmt.Sprintf(":%v", *httpPort), n)
+	serviceWait := sync.WaitGroup{}
```

Comment:
I might make this a pointer to a wg, just to avoid accidentally copying it later.